routes/profiles.rb

```diff
@@ -50,13 +51,29 @@ def self.registered(app)
           end
 
           update_profile = lambda do
-            if !@profile || @profile[:role] != 'superadmin'
-              halt 400, 'Parameter requirements were not met.' unless check_parameters(params, profile_params)
+            updatedProfile = params[:updatedProfile]
+            if !@profile || @profile[:id].to_s != updatedProfile[:_id][:$oid].to_s
```

Comment:
Is this going to blow up if a superadmin tries to edit someone else's account?

Reply:
That's a good question. I haven't been testing with a superadmin account. I was hoping the inner if would handle the bypass for superadmin accounts. But I'll need to test it a little more to verify. I am just not very familiar with where the superadmin actions are on the app.